core/db/migrate/20151021163309_convert_sale_promotions.rb

```diff
@@ -0,0 +1,28 @@
+class ConvertSalePromotions < ActiveRecord::Migration
+  def up
+    sale_promotions.update_all(apply_automatically: true)
+  end
+
+  def down
+    # intentionally left blank
+  end
+
+  private
+
+  def sale_promotions
+    promo_table = Spree::Promotion.arel_table
+    code_table  = Spree::PromotionCode.arel_table
+
+    promotion_code_join = promo_table.join(code_table, Arel::Nodes::OuterJoin).on(
+      promo_table[:id].eq(code_table[:promotion_id])
+    ).join_sources
+
+    Spree::Promotion.includes(:promotion_rules).
+      joins(promotion_code_join).
+      where(
+        code_table[:value].eq(nil).and(
+          promo_table[:path].eq(nil)
+        )
+      ).distinct
+  end
```

Comment:
This is pretty much the old `Spree::PromotionHandler::Cart#sale_promtions` method - I'm sure it could be written better but given it's for a migration I thought it was best to just copy it across as it was.